### ISSUE TITLE
sorted_imports rule is not caring about different import flavors

### ISSUE THREAD

**Issue by m-revetria** [realm/SwiftLint#1269]
Imports can declare which kind of type wants to imports, see [Import Declaration from Apple Docs](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Declarations.html#//apple_ref/doc/uid/TP40014097-CH34-ID354). Additionally imports could be marked with `@testable`. 

Personally I'd like having `@testable` at the end of imports list and grouping imports with `kind` declaration by `kind` after normal imports 

1. I would expect that next case don't throw. 
   ```swift
   import UIKit
   @testable import MyApp
   ```

2. I would expect that next case throws
    ```swift
    import AVKit
    import struct MyApp.MyStruct
    import enum MyApp.MyEnum
    import class Foundation.NSArray
    import UIKit
    ```

3. Next case would be the valid version of previous case. Actually it is not throwing because `sorted_imports` rule is case sensitive. I would also expect it to be case insensitive as pointed out in this issue [#1185](https://github.com/realm/SwiftLint/issues/1185)
    ```swift
    import AVKit
    import UIKit
    import class Foundation.NSArray
    import class MyApp.MyClass
    import enum MyApp.MyEnum
    import struct MyApp.MyStruct
    ```

4. I would expect next case don't throw
    ```swift
    import AVKit
    import test
    import class Foundation.NSArray
    @testable import MyApp
    ```
